module/zfs/zfs_znode.c

```diff
@@ -1049,6 +1075,11 @@ zfs_xvattr_set(znode_t *zp, xvattr_t *xvap, dmu_tx_t *tx)
 		    zp->z_pflags, tx);
 		XVA_SET_RTN(xvap, XAT_SPARSE);
 	}
+	if (XVA_ISSET_REQ(xvap, XAT_PROJINHERIT)) {
+		ZFS_ATTR_SET(zp, ZFS_PROJINHERIT, xoap->xoa_projinherit,
+		    zp->z_pflags, tx);
```

Comment:
@Nasf-Fan I agree with your assessment of the semantics required to enforce PROJINHERIT.  But if we aren't going to enforce it, what use is it?  How would userland or the administrator enforce proper usage?  I am getting the feeling that PROJINHERIT / tree quota is not fully thought out at this point in time.  I think we need a more complete design, or we should drop it from the PR.